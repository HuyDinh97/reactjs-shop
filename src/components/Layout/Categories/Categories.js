import React from 'react';
import Nav from 'react-bootstrap/Nav';
import useFetchCategory from 'Hooks/useFetchCategory';

import classes from './Categories.module.css';

function Categories() {
  const { categories, isSuccess } = useFetchCategory(); // Destructuring

  // Safe code: handle if API error in response
  if (isSuccess === false) {
    return <p data-testid="error-fetch">Error, cannot fetch API</p>;
  }

  if (!categories) {
    return <p>Loading...</p>;
  }

  return (
    <Nav defaultActiveKey="/home" as="ul" data-testid="category-element">
      {categories &&
        categories.map((category) => (
          <Nav.Item key={category._id} as="li">
            <Nav.Link className={classes.nav_categories} href="/">
              {category.name}
            </Nav.Link>
          </Nav.Item>
        ))}
    </Nav>
  );
}

export default Categories;
