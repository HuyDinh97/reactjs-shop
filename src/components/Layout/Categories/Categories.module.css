.nav_categories {
  color: black !important;
  font-weight: 600 !important;
  font-size: smaller !important;
}