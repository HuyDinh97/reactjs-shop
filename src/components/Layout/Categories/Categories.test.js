import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Categories from './Categories';

test('should Header render correctly', () => {
  const Wrapper = render(<Categories />);
  expect(Wrapper).toMatchSnapshot();
});
