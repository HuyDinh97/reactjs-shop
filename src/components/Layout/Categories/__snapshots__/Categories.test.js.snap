// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Header render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link active"
            data-rr-ui-event-key="/home"
            href="/home"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="link-1"
            href="#"
            role="button"
            tabindex="0"
          >
            Fashion Sneaker
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rr-ui-event-key="link-2"
            href="#"
            role="button"
            tabindex="0"
          >
            Jackets
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rr-ui-event-key="/home"
          href="/home"
        >
          Home
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="link-1"
          href="#"
          role="button"
          tabindex="0"
        >
          Fashion Sneaker
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="link-2"
          href="#"
          role="button"
          tabindex="0"
        >
          Jackets
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
